api/api_ethereum.go

```diff
@@ -502,52 +503,228 @@ type EthRPCTransaction struct {
 	S                *hexutil.Big    `json:"s"`
 }
 
+// newEthRPCTransactionFromBlockAndIndex creates an EthRPCTransaction from block and index parameters.
+func newEthRPCTransactionFromBlockAndIndex(b *types.Block, index uint64) (*EthRPCTransaction, error) {
+	txs := b.Transactions()
+	if index >= uint64(len(txs)) {
+		return nil, errors.New("invalid transaction index")
+	}
```

Comment:
@yoomee1313 
Hi yumiel :) 
I saw your [metion](https://github.com/klaytn/klaytn/pull/1087#discussion_r773330714) about handling error.

Can you check [here](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR583-R586) and [here](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR595-R598) too?

When error occurred, it returns nil based on those checking logic.

Reply:
I think returning error would be good because 
someday we can add some additional logic based on this error (for debugging or logging purpose, etc...)

Reply:
@jimni1222 
Hi jamie. 
I'd like to hear your opinion about the [comment](https://github.com/klaytn/klaytn/pull/1087#discussion_r773330714) of @yoomee1313 .

**Comment descriptoin:**
yoomee said that "[newEthTransactionFromBlockAndIndex](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR506-R507) return error but there is no actual handling logic about it, so removing error and logging instead would be great option"

I checked [newEthRPCTransaction](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR530-R531) also and removing error seems to be ok because it always return `nil`.

We need to discussion about which option would be better.
cc. @yoomee1313

Reply:
@jimni1222 @yoomee1313 
I applied yoomee's suggestions [here](https://github.com/klaytn/klaytn/pull/1093/commits/dfb10afd4daa7bbaa14fdb826b1968181783f7e2).

Please review this commit :)

When error happened, it logs like below and does not return error.
```
ERROR[01/05,15:26:43 +09] [3] invalid transaction index                 given index=1 length of txs=1
```

FYI.
Both eth and klay namespace Tx retrieving apis returns just nil data when something goes wrong.

e.g.
**Retrieve non-existing transaction using `eth` namespace.**
```shell
curl http://localhost:8551 \
-X POST \
-H "Content-Type: application/json" \
-d '{"jsonrpc":"2.0","method":"eth_getTransactionByBlockHashAndIndex","params":["0x4c632f9774638ba3ab06095b5c9eb3b9ef42381653dee821e18f234168e0b975", "0x0"],"id":1}'
{"jsonrpc":"2.0","id":1,"result":null}
```
**Retrieve non-existing transaction using `klay` namespace.**
```shell
$ curl http://localhost:8551 \
-X POST \
-H "Content-Type: application/json" \
-d '{"jsonrpc":"2.0","method":"klay_getTransactionByBlockHashAndIndex","params":["0x4c632f9774638ba3ab06095b5c9eb3b9ef42381653dee821e18f234168e0b975", "0x3"],"id":1}'
{"jsonrpc":"2.0","id":1,"result":null}
```